spine-c/src/spine/SkeletonJson.c

```diff
@@ -680,14 +725,10 @@ spSkeletonData* spSkeletonJson_readSkeletonData (spSkeletonJson* self, const cha
 					spAttachmentType type;
 					if (strcmp(typeString, "region") == 0)
 						type = SP_ATTACHMENT_REGION;
-					else if (strcmp(typeString, "mesh") == 0)
+					else if (strcmp(typeString, "mesh") == 0 || strcmp(typeString, "weightedmesh") == 0 || strcmp(typeString, "skinnedmesh") == 0)
 						type = SP_ATTACHMENT_MESH;
-					else if (strcmp(typeString, "weightedmesh") == 0 || strcmp(typeString, "skinnedmesh") == 0)
-						type = SP_ATTACHMENT_WEIGHTED_MESH;
-					else if (strcmp(typeString, "linkedmesh") == 0)
+					else if (strcmp(typeString, "linkedmesh") == 0 || strcmp(typeString, "weightedlinkedmesh") == 0)
```

Comment:
I think it's safe to not check for the old types now: weightemesh, skinnedmesh, weightedlinkedmesh.